sensor: Update wsen_itds driver/sample to new DTS based APIs

galak opened the pull request.

galak committed:
drivers: sensor: wsen_itds: Update driver to use i2c_dt_spec

Move driver to use i2c_dt_spec for I2C bus access.

Signed-off-by: Kumar Gala <galak@kernel.org>

galak committed:
samples: wsen_itds: Convert sample to use DEVICE_DT_GET_ONE

Move to use DEVICE_DT_GET_ONE instead of device_get_binding as we
work on phasing out use of DTS 'label' property.

Signed-off-by: Kumar Gala <galak@kernel.org>

gmarull commented on drivers/sensor/wsen_itds/itds.h:
```diff
@@ -91,8 +91,7 @@ struct itds_accel_range {
 struct itds_device_config {
-	const char *bus_name;
-	uint16_t i2c_addr;
+	const struct i2c_dt_spec i2c;
```
`const` is redundant here, the whole config is already defined as `const`.

gmarull commented on drivers/sensor/wsen_itds/itds.c:
```diff
@@ -385,17 +385,23 @@ static const struct sensor_driver_api itds_api = {
-	.int_flags = DT_INST_GPIO_FLAGS(idx, int_gpios),		\
 	.def_odr = DT_INST_ENUM_IDX(idx, odr),				\
 	.def_op_mode = DT_INST_ENUM_IDX(idx, op_mode),			\
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(idx, int_gpios),		\
+		(WSEN_ITDS_CFG_IRQ(idx)), ())				\
```
suggestion:

```c
IF_ENABLED(CONFIG_ITDS_TRIGGER, (.int_gpio = GPIO_DT_SPEC_INST_GET(inst, int_gpios),))
```

If int_gpio is optional even with `CONFIG_ITDS_TRIGGER=y` use `GPIO_DT_SPEC_INST_GET_OR` and adjust init code accordingly.

galak replied:
Update the code here, I went with `GPIO_DT_SPEC_INST_GET_OR` since we could have multiple instances of the device and some support / have `int_gpio` in dts I didn't consider it an error if the prop was missing.

gmarull commented on drivers/sensor/wsen_itds/itds.h:
```diff
@@ -92,17 +92,16 @@ struct itds_accel_range {
-	gpio_pin_t int_pin;
-	gpio_dt_flags_t int_flags;
+#ifdef	CONFIG_ITDS_TRIGGER
+	const struct gpio_dt_spec int_gpio;
```
`const` is redundant here

gmarull commented on samples/sensor/wsen_itds/src/main.c:
```diff
@@ -131,9 +131,9 @@ void main(void)
-	itds = device_get_binding(DT_LABEL(DT_INST(0, we_wsen_itds)));
-	if (!itds) {
-		printf("Device not found.\n");
+	itds = DEVICE_DT_GET_ONE(we_wsen_itds);
```
suggestion: itds can be initialized at L130

galak committed:
drivers: sensor: wsen_itds: Update driver to use gpio_dt_spec

Move driver to use gpio_dt_spec for GPIO interrupt handling.  Since
the driver can support multiple instances its possible that some
have irq's in dts and some don't.  So we add a check in
itds_trigger_set to make sure we have a GPIO device if we are
trying to set a trigger.

Signed-off-by: Kumar Gala <galak@kernel.org>

gmarull commented on drivers/sensor/wsen_itds/itds_trigger.c:
```diff
@@ -96,24 +96,29 @@ int itds_trigger_mode_init(const struct device *dev)
+	/* dts doesn't have GPIO int pin set, so we dont support
+	 * trigger mode for this instance
+	 */
+	if (!cfg->int_gpio.port) {
+		return 0;
+	}
```
the API trigger set should also return -ENOTSUP if this holds true (L41)

galak replied:
I'm not sure that makes sense in this case.  As the caller `itds_init` is just going to ignore the error code.

gmarull replied:
If we don't do that, application code may try to configure a trigger on an instance that doesn't support it, right?

galak replied:
Added check in `itds_trigger_set`
